cms/djangoapps/contentstore/management/commands/replace_v1_lib_refs_with_v2_in_courses.py

```diff
@@ -0,0 +1,71 @@
+"""
+A Command which, given a mapping of V1 to V2 Libraries,
+edits all xblocks in courses which refer to the v1 library to point to the v2 library.
+"""
+
+import logging
+import csv
+
+from django.core.management import BaseCommand, CommandError
+from celery import group
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore import ModuleStoreEnum
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from cms.djangoapps.contentstore.tasks import replace_all_library_source_blocks_ids_for_course
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py cms replace_v1_lib_refs_with_v2_in_courses '/path/to/file/containing/library_mappings.csv'
+        $ ./manage.py cms replace_v1_lib_refs_with_v2_in_courses '/path/to/file/containing/library_mappings.csv'
+
+    """
+    def add_arguments(self, parser):
+        parser.add_argument('file_path', type=str, help='Path to the CSV file.')
+
+    def replace_all_library_source_blocks_ids(self, v1_to_v2_lib_map):
+        """A method to replace 'source_library_id' in all relevant blocks."""
```

Comment:
It's not strictly necessary, but some checking on the library map seems appropriate (e.g., all input and output libraries referenced in the map exist; the number of existing libraries is twice the number of rows in the mapping).

I presume that this mapping is automatically generated by a management command run before this one. The check is a lightweight operaiton to make sure no corruption occurred in between.